### DISCUSSION TITLE
Can't mogrify jpg to jxl

### DISCUSSION THREAD

**Discussion by coltEli** [ImageMagick/ImageMagick#7659] [Help] [answered]
### ImageMagick version

7.1.1-39

### Operating system, version and so on

Debian Bookworm

### Description

Hello, I am trying to batch convert jpg to jxl using mogrify. Getting this error though.

`mogrify -format jxl in.jpg
mogrify: unable to open image 'jxl:': No such file or directory @ error/blob.c/OpenBlob/3596.
mogrify: unrecognized image format jxl' @ error/mogrify.c/MogrifyImageCommand/4950.`

this works, so seems problem with jxl format
`mogrify -format png in.jpg`

but i can convert directly
`magick in.jpg out.jxl`

Any ideas on why this is?

**Comment by fmw42** [comment]
Syntax is different for IM 7 than IM 6.  On IM 7, use magick mogrify, not mogrify.  (But use magick, not magick convert and not convert). See https://imagemagick.org/script/porting.php

**Comment by dlemstra** [comment, marked as answer]
Did you check the output file that was produced with this command: `magick in.jpg out.jxl`? I suspect your output is a jpeg image and that you don't have support for jpeg-xl in your build. Is it enabled in `magick -list format`?

**Reply by coltEli** [reply]
This seems likely. I am not sure how to tell if it truly is jxl, but it did create a viewable file with no feedback that something was amiss. The resultant file is 3% smaller going along with your theory that it is simply a jpg compression of the original.  

I then installed cjxl and ran `cjxl in.jpg out.jxl` and this resulted  in significant savings of 27%.  Unfortunately this strips the metadata.

`magick -list format` had no reference to jxl. I will have to read up on how to include jxl in the build so I can use mogrify to easily retain the metadata.
